### ISSUE TITLE
HttpContext is null in CookieAuthenticationService.cs when installing a plugin

### ISSUE THREAD

**Issue by ZachIsAGardner** [nopSolutions/nopCommerce#4089]
In nopCommerce 4.20 I've written a custom plugin and when installing the plugin on startup I get a null object reference error in CookieAuthenticationService.cs.

Happens here in GetAuthenticatedCustomer:
```
//try to get authenticated user identity
var authenticateResult = _httpContextAccessor.HttpContext.AuthenticateAsync(NopAuthenticationDefaults.AuthenticationScheme).Result;
if (!authenticateResult.Succeeded)
    return null;
```

I've replaced it with this for now in the source code:
```
AuthenticateResult authenticateResult = _httpContextAccessor?.HttpContext?.AuthenticateAsync(NopAuthenticationDefaults.AuthenticationScheme)?.Result;
if (authenticateResult == null || !authenticateResult.Succeeded)
    return null;
```
Why is this happening?

**Comment by RomanovM** [comment]
See also: https://www.nopcommerce.com/boards/t/70453/plugin-uninstall-then-install-problem-in-nopcommerce-420.aspx

**Comment by RomanovM** [comment]
Explanation of the error and ways to solve it here https://www.nopcommerce.com/boards/t/72546/errors-on-plugin-install-in-nopcommerce-420.aspx

**Comment by exileDev** [comment]
Closed #4089
